crates/iceberg/src/error.rs

```diff
@@ -44,6 +44,14 @@ pub enum ErrorKind {
     ///
     /// This error is returned when given iceberg feature is not supported.
     FeatureUnsupported,
+    /// Could not cast a byte slice to appropriate array
+    ///
+    /// This error occurs when converting Bytes to iceberg values.
+    ByteArrayConversionFailed,
```

Comment:
I think those kind could be treated as invalid data.

Reply:
Yes, I also think that maybe can treat as a more general error kind.🤔